scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/LiveJobs.java

```diff
@@ -574,17 +574,18 @@ public TerminationData taskTerminatedWithResult(TaskId taskId, TaskResultImpl re
                 } else if (numberOfExecutionLeft > 0) {
                     logger.info("Number of execution left is " + numberOfExecutionLeft);
 
-                    if (onErrorPolicyInterpreter.requiresPauseTaskOnError(task)) {
-                        suspendTaskOnError(jobData, task, result.getTaskDuration());
-
-                        logger.info("Task is paused on error");
+                    if (onErrorPolicyInterpreter.requiresPauseTaskOnError(task) || requiresPauseJobOnError) {
 
-                        return terminationData;
-                    } else if (requiresPauseJobOnError) {
-                        suspendTaskOnError(jobData, task, result.getTaskDuration());
-                        pauseJob(task.getJobId());
+                        long waitTime = jobData.job.getNextWaitingTime(task.getMaxNumberOfExecution() -
+                                                                       numberOfExecutionLeft);
+                        restartTaskOnError(jobData,
+                                           task,
+                                           TaskStatus.WAITING_ON_ERROR,
+                                           result,
+                                           waitTime,
+                                           terminationData);
 
-                        logger.info("Job is paused on error");
+                        logger.info("New restart is scheduled");
```

Comment:
Please append more information to the log message so that it is possible to identify what is the Task that is scheduled for a restart (e.g. the Task ID).